gstools/krige/base.py

```diff
@@ -122,24 +165,35 @@ def __call__(
         x, y, z, self.pos, __, mt_changed, axis_lens = self._pre_pos(
             pos, mesh_type, make_unstruct=True
         )
-        point_no = len(x)
-        # set chunk size
-        chunk_size = point_no if chunk_size is None else int(chunk_size)
-        chunk_no = int(np.ceil(point_no / chunk_size))
         field = np.empty_like(x)
         krige_var = np.empty_like(x)
-        ext_drift = self._pre_ext_drift(point_no, ext_drift)
-        # iterate of chunks
-        for i in range(chunk_no):
-            # get chunk slice for actual chunk
-            chunk_slice = (i * chunk_size, min(point_no, (i + 1) * chunk_size))
-            c_slice = slice(*chunk_slice)
-            # get RHS of the kriging system
-            k_vec = self._get_krige_vecs((x, y, z), chunk_slice, ext_drift)
-            # generate the raw kriging field and error variance
-            field[c_slice], krige_var[c_slice] = krigesum(
-                self._krige_mat, k_vec, self._krige_cond
-            )
+        # set constant mean if present and wanted
+        if only_mean and self.has_const_mean:
+            field[...] = self.get_mean() - self.mean  # mean is added later
+            krige_var[...] = self.model.sill  # will result in 0 var. later
+        # execute the kriging routine
+        else:
+            point_no = len(x)
+            # set chunk size
+            chunk_size = point_no if chunk_size is None else int(chunk_size)
+            chunk_no = int(np.ceil(point_no / chunk_size))
+            ext_drift = self._pre_ext_drift(point_no, ext_drift)
+            # iterate of chunks
+            for i in range(chunk_no):
+                # get chunk slice for actual chunk
+                chunk_slice = (
+                    i * chunk_size,
+                    min(point_no, (i + 1) * chunk_size),
+                )
+                c_slice = slice(*chunk_slice)
+                # get RHS of the kriging system
+                k_vec = self._get_krige_vecs(
+                    (x, y, z), chunk_slice, ext_drift, only_mean
+                )
+                # generate the raw kriging field and error variance
+                field[c_slice], krige_var[c_slice] = krigesum(
+                    self._krige_mat, k_vec, self._krige_cond
+                )
```

Comment:
This loop could be refactored into its own private method for readability.

Reply:
I think, I'll leave it this way, because a separate routine would have a really long call signature at this point. I can live with the 10 extra lines ;-)